pyclibrary/c_parser.py

```diff
@@ -527,26 +523,27 @@ def pa(t):
                     if_hit.append(ev)
 
                 elif d == 'define':
-                    if not if_true[-1]:
-                        continue
-                    logger.debug("  "*(len(if_true)-1) + "define: " +
-                                 '{}, {}'.format(macroName, rest))
-                    try:
-                        # Macro is registered here
-                        self.pp_define.parseString(macroName + ' ' + rest)
-                    except Exception:
-                        logger.exception("Error processing macro definition:" +
-                                         '{}, {}'.format(macroName, rest))
+                    if if_true[-1]:
+                        logger.debug("  "*(len(if_true)-1) + "define: " +
+                                     '{}, {}'.format(macro_name, rest))
+                        try:
+                            # Macro is registered here
+                            self.pp_define.parseString(
+                                macro_name + ' ' + rest)
+                        except Exception:
+                            logger.exception(
+                                'Error processing macro definition:{}, {}'
+                                .format(macro_name, rest))
 
                 elif d == 'undef':
-                    if not if_true[-1]:
-                        continue
-                    try:
-                        self.clib_intf.del_macro(macroName.strip())
-                    except Exception:
-                        if sys.exc_info()[0] is not KeyError:
-                            mess = "Error removing macro definition '{}'"
-                            logger.exception(mess.format(macroName.strip()))
+                    if if_true[-1]:
+                        try:
+                            self.clib_intf.del_macro(macro_name.strip())
+                        except Exception:
+                            if sys.exc_info()[0] is not KeyError:
```

Comment:
Of course!!!
Don't know how I could oversee that code smell when modifying the code.